internal/pkg/term/selector/selector.go

```diff
@@ -778,27 +787,84 @@ func (s *LocalWorkloadSelector) Job(msg, help string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("retrieve jobs from workspace: %w", err)
 	}
-	storeJobNames, err := s.ConfigSelector.workloadLister.ListJobs(summary.Application)
+	storeJobs, err := s.ConfigSelector.workloadLister.ListJobs(summary.Application)
 	if err != nil {
 		return "", fmt.Errorf("retrieve jobs from store: %w", err)
 	}
-	jobNames := filterWlsByName(storeJobNames, wsJobNames)
-	if len(jobNames) == 0 {
-		return "", errors.New("no jobs found")
-	}
-	if len(jobNames) == 1 {
-		log.Infof("Only found one job, defaulting to: %s\n", color.HighlightUserInput(jobNames[0]))
-		return jobNames[0], nil
+	options, err := s.getWorkloadSelectOptions(storeJobs, wsJobNames, jobWorkloadType)
+	if err != nil {
+		return "", err
 	}
 
-	selectedJobName, err := s.prompt.SelectOne(msg, help, jobNames, prompt.WithFinalMessage(jobNameFinalMsg))
+	if len(options) == 1 {
+		log.Infof("Found only one job, defaulting to: %s\n", color.HighlightUserInput(options[0].Value))
+		return options[0].Value, nil
+	}
+	selectedJobName, err := s.prompt.SelectOption(msg, help, options, prompt.WithFinalMessage(workloadFinalMsg))
 	if err != nil {
 		return "", fmt.Errorf("select job: %w", err)
 	}
 	return selectedJobName, nil
+
+}
+
+func (s *LocalWorkloadSelector) getWorkloadSelectOptions(storeWls []*config.Workload, wsWlNames []string, workloadType string) ([]prompt.Option, error) {
+	var pluralNounString string
+	switch workloadType {
+	case anyWorkloadType:
+		pluralNounString = "jobs or services"
+	case svcWorkloadType:
+		fallthrough
+	case jobWorkloadType:
+		pluralNounString = english.PluralWord(2, workloadType, "")
+	default:
+		return nil, fmt.Errorf("unrecognized workload type %q", workloadType)
+	}
+
+	var options []prompt.Option
+
+	// Get the list of initialized workloads that are present in the workspace
+	initializedLocalWorkloads := filterWlsByName(storeWls, wsWlNames)
+	for _, wl := range initializedLocalWorkloads {
+		options = append(options, prompt.Option{Value: wl})
+	}
+
+	// Return early if we're only looking for config store workloads.
+	if !s.allLocalWorkloads {
```

Comment:
I feel like `allLocalWorkloads` should be `true` by default for `LocalWorkloadSelector`. Maybe consider renaming the var as `WithOnlyInitialized`? And our existing callers (who wish to keep the current behavior to only include initialized workloads) for `LocalWorkloadSelector`'s methods will set `WithOnlyInitialized` to `true`.

Reply:
Done! I flipped the behavior.